main/ui/src/main/java/org/cryptomator/ui/changepassword/ChangePasswordController.java

```diff
@@ -23,32 +25,41 @@
 import javafx.stage.Stage;
 import java.io.IOException;
 import java.nio.CharBuffer;
-import java.util.Optional;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.security.SecureRandom;
 
 import static org.cryptomator.common.Constants.MASTERKEY_FILENAME;
 
 @ChangePasswordScoped
 public class ChangePasswordController implements FxController {
 
 	private static final Logger LOG = LoggerFactory.getLogger(ChangePasswordController.class);
+	private static final String MASTERKEY_BACKUP_SUFFIX = ".bkup";
```

Comment:
Is the file ending used elsewhere in the application? Might be worth deduplicating it then by moving it to the `Constants` class.